### ISSUE TITLE
Cannot transcode media, YUV2MPEG error

### ISSUE THREAD

**Issue by gidoBOSSftw5731** [master-of-zen/Av1an#604]
Error:
```
ERROR [av1an_core::broker] [chunk 153] encoder crashed 3 times, shutting down worker
ERROR [av1an_core::broker] [chunk 153] encoder crashed: exit status: 1
stdout:

stderr:
        Error parsing header; not a YUV2MPEG2 file?
        Fatal: Unsupported Y4M stream.


source pipe stderr:

ffmpeg pipe stderr:
```

Command: 
```
$ av1an -i 'file.mkv' -e vpx -w 12 -a " -c:a copy -map 0" -v " --cq-level=30 --end-usage=cq --threads=1 " -o 'out.mkv'
```

ffprobe:

> ```
> ffprobe version 4.4 Copyright (c) 2007-2021 the FFmpeg developers
>   built with gcc 10 (Debian 10.2.1-6)
>   configuration: --disable-decoder=amrnb --disable-decoder=libopenjpeg --disable-gnutls --disable-liblensfun --disable-libopencv --disable-podpages --disable-sndio --disable-stripping --enable-avfilter --enable-avresample --enable-gcrypt --enable-gpl --enable-ladspa --enable-libaom --enable-libaribb24 --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libdavs2 --enable-libdc1394 --enable-libfdk-aac --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libiec61883 --enable-libjack --enable-libkvazaar --enable-libmp3lame --enable-libmysofa --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libtesseract --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libwebp --enable-libx265 --enable-libxavs2 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-nonfree --enable-openal --enable-opencl --enable-opengl --enable-openssl --enable-postproc --enable-pthreads --enable-shared --enable-version3 --enable-vulkan --incdir=/usr/include/x86_64-linux-gnu --libdir=/usr/lib/x86_64-linux-gnu --prefix=/usr --toolchain=hardened --enable-frei0r --enable-chromaprint --enable-libx264 --enable-vaapi --enable-libmfx --enable-libvmaf --enable-libsvtav1 --enable-libilbc --disable-altivec --shlibdir=/usr/lib/x86_64-linux-gnu
>   libavutil      56. 70.100 / 56. 70.100
>   libavcodec     58.134.100 / 58.134.100
>   libavformat    58. 76.100 / 58. 76.100
>   libavdevice    58. 13.100 / 58. 13.100
>   libavfilter     7.110.100 /  7.110.100
>   libavresample   4.  0.  0 /  4.  0.  0
>   libswscale      5.  9.100 /  5.  9.100
>   libswresample   3.  9.100 /  3.  9.100
>   libpostproc    55.  9.100 / 55.  9.100
> [matroska,webm @ 0x556b153dcf80] Could not find codec parameters for stream 5 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
> Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
> [matroska,webm @ 0x556b153dcf80] Could not find codec parameters for stream 6 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
> Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
> [matroska,webm @ 0x556b153dcf80] Could not find codec parameters for stream 7 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
> Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
> [matroska,webm @ 0x556b153dcf80] Could not find codec parameters for stream 8 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
> Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
> [matroska,webm @ 0x556b153dcf80] Could not find codec parameters for stream 9 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
> Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
> Input #0, matroska,webm, from 'file.mkv':
>   Metadata:
>     title           : legitmedia
>     encoder         : libebml v1.3.4 + libmatroska v1.4.5
>     creation_time   : 2016-09-13T20:43:41.000000Z
>   Duration: 00:47:12.70, start: 0.000000, bitrate: 80277 kb/s
>   Stream #0:0(eng): Video: h264 (High), yuv420p(progressive), 3840x2160 [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 1k tbn, 47.95 tbc (default)
>     Metadata:
>       title           : legitmedia
>       BPS             : 74132620
>       BPS-eng         : 74132620
>       DURATION        : 00:47:12.663000000
>       DURATION-eng    : 00:47:12.663000000
>       NUMBER_OF_FRAMES: 67916
>       NUMBER_OF_FRAMES-eng: 67916
>       NUMBER_OF_BYTES : 26249091355
>       NUMBER_OF_BYTES-eng: 26249091355
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:1(eng): Audio: dts (DTS-HD MA), 48000 Hz, 5.1(side), s16p (default)
>     Metadata:
>       title           : DTS-HD MA 5.1 @ 1825 kbps - Blu-ray
>       BPS             : 1824571
>       BPS-eng         : 1824571
>       DURATION        : 00:47:12.672000000
>       DURATION-eng    : 00:47:12.672000000
>       NUMBER_OF_FRAMES: 265563
>       NUMBER_OF_FRAMES-eng: 265563
>       NUMBER_OF_BYTES : 646051724
>       NUMBER_OF_BYTES-eng: 646051724
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:2(rus): Audio: dts (DTS-HD MA), 48000 Hz, 5.1(side), s16p
>     Metadata:
>       title           : DTS-HD MA 5.1 @ 1773 kbps - MVO, LostFilm
>       BPS             : 1772944
>       BPS-eng         : 1772944
>       DURATION        : 00:47:12.704000000
>       DURATION-eng    : 00:47:12.704000000
>       NUMBER_OF_FRAMES: 265566
>       NUMBER_OF_FRAMES-eng: 265566
>       NUMBER_OF_BYTES : 627778212
>       NUMBER_OF_BYTES-eng: 627778212
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:3(rus): Audio: dts (DTS-HD MA), 48000 Hz, 5.1(side), s16p
>     Metadata:
>       title           : DTS-HD MA 5.1 @ 1780 kbps - DVO, KvK
>       BPS             : 1779988
>       BPS-eng         : 1779988
>       DURATION        : 00:47:12.704000000
>       DURATION-eng    : 00:47:12.704000000
>       NUMBER_OF_FRAMES: 265566
>       NUMBER_OF_FRAMES-eng: 265566
>       NUMBER_OF_BYTES : 630272620
>       NUMBER_OF_BYTES-eng: 630272620
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:4(rus): Audio: ac3, 48000 Hz, 5.1(side), fltp, 640 kb/s
>     Metadata:
>       title           : AC3 5.1 @ 640 kbps - AVO
>       BPS             : 640000
>       BPS-eng         : 640000
>       DURATION        : 00:47:12.704000000
>       DURATION-eng    : 00:47:12.704000000
>       NUMBER_OF_FRAMES: 88522
>       NUMBER_OF_FRAMES-eng: 88522
>       NUMBER_OF_BYTES : 226616320
>       NUMBER_OF_BYTES-eng: 226616320
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:5(eng): Subtitle: hdmv_pgs_subtitle
>     Metadata:
>       BPS             : 14092
>       BPS-eng         : 14092
>       DURATION        : 00:40:05.779000000
>       DURATION-eng    : 00:40:05.779000000
>       NUMBER_OF_FRAMES: 840
>       NUMBER_OF_FRAMES-eng: 840
>       NUMBER_OF_BYTES : 4237925
>       NUMBER_OF_BYTES-eng: 4237925
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:6(ara): Subtitle: hdmv_pgs_subtitle
>     Metadata:
>       BPS             : 10312
>       BPS-eng         : 10312
>       DURATION        : 00:40:05.779000000
>       DURATION-eng    : 00:40:05.779000000
>       NUMBER_OF_FRAMES: 844
>       NUMBER_OF_FRAMES-eng: 844
>       NUMBER_OF_BYTES : 3101154
>       NUMBER_OF_BYTES-eng: 3101154
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:7(fre): Subtitle: hdmv_pgs_subtitle
>     Metadata:
>       BPS             : 9763
>       BPS-eng         : 9763
>       DURATION        : 00:42:56.700000000
>       DURATION-eng    : 00:42:56.700000000
>       NUMBER_OF_FRAMES: 832
>       NUMBER_OF_FRAMES-eng: 832
>       NUMBER_OF_BYTES : 3144579
>       NUMBER_OF_BYTES-eng: 3144579
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:8(ger): Subtitle: hdmv_pgs_subtitle
>     Metadata:
>       BPS             : 14714
>       BPS-eng         : 14714
>       DURATION        : 00:40:05.779000000
>       DURATION-eng    : 00:40:05.779000000
>       NUMBER_OF_FRAMES: 840
>       NUMBER_OF_FRAMES-eng: 840
>       NUMBER_OF_BYTES : 4425038
>       NUMBER_OF_BYTES-eng: 4425038
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:9(tur): Subtitle: hdmv_pgs_subtitle
>     Metadata:
>       BPS             : 12933
>       BPS-eng         : 12933
>       DURATION        : 00:40:05.779000000
>       DURATION-eng    : 00:40:05.779000000
>       NUMBER_OF_FRAMES: 840
>       NUMBER_OF_FRAMES-eng: 840
>       NUMBER_OF_BYTES : 3889521
>       NUMBER_OF_BYTES-eng: 3889521
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:10(eng): Subtitle: hdmv_pgs_subtitle, 1920x1080
>     Metadata:
>       BPS             : 24480
>       BPS-eng         : 24480
>       DURATION        : 00:47:08.284000000
>       DURATION-eng    : 00:47:08.284000000
>       NUMBER_OF_FRAMES: 1666
>       NUMBER_OF_FRAMES-eng: 1666
>       NUMBER_OF_BYTES : 8654724
>       NUMBER_OF_BYTES-eng: 8654724
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:11(fre): Subtitle: hdmv_pgs_subtitle, 1920x1080
>     Metadata:
>       BPS             : 17526
>       BPS-eng         : 17526
>       DURATION        : 00:47:08.284000000
>       DURATION-eng    : 00:47:08.284000000
>       NUMBER_OF_FRAMES: 1636
>       NUMBER_OF_FRAMES-eng: 1636
>       NUMBER_OF_BYTES : 6196121
>       NUMBER_OF_BYTES-eng: 6196121
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>   Stream #0:12(ger): Subtitle: hdmv_pgs_subtitle, 1920x1080
>     Metadata:
>       BPS             : 25353
>       BPS-eng         : 25353
>       DURATION        : 00:47:08.284000000
>       DURATION-eng    : 00:47:08.284000000
>       NUMBER_OF_FRAMES: 1676
>       NUMBER_OF_FRAMES-eng: 1676
>       NUMBER_OF_BYTES : 8963348
>       NUMBER_OF_BYTES-eng: 8963348
>       _STATISTICS_WRITING_APP: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_APP-eng: mkvmerge v9.4.0 ('Knurl') 64bit
>       _STATISTICS_WRITING_DATE_UTC: 2016-09-13 20:43:41
>       _STATISTICS_WRITING_DATE_UTC-eng: 2016-09-13 20:43:41
>       _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
>       _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
> ```

Version:
```
$ av1an -V
av1an 0.3.1-unstable (rev 154a914) (release)

* Compiler
  rustc 1.59.0 (LLVM 13.0)

* Target Triple
  x86_64-unknown-linux-gnu

* Date Info
   Build Date:  2022-03-31
  Commit Date:  2022-03-29
```

**Comment by redzic** [comment]
Could you check if any other encoder besides vpx works for you for this video? Also, do you have lsmash or ffsm2 installed?

**Comment by gidoBOSSftw5731** [comment]
Apologies for the delay, does not work for AOM either:
```
$ av1an -i 'file.mkv' -e aom --split-method none -w 12 -o 'out.mkv'
[matroska,webm @ 0x55892a3fc500] Could not find codec parameters for stream 5 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a3fc500] Could not find codec parameters for stream 6 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a3fc500] Could not find codec parameters for stream 7 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a3fc500] Could not find codec parameters for stream 8 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a3fc500] Could not find codec parameters for stream 9 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a411500] Could not find codec parameters for stream 5 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a411500] Could not find codec parameters for stream 6 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a411500] Could not find codec parameters for stream 7 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a411500] Could not find codec parameters for stream 8 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a411500] Could not find codec parameters for stream 9 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a3f9800] Could not find codec parameters for stream 5 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a3f9800] Could not find codec parameters for stream 6 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a3f9800] Could not find codec parameters for stream 7 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a3f9800] Could not find codec parameters for stream 8 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55892a3f9800] Could not find codec parameters for stream 9 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
INFO [av1an_core::settings] scenecut: found 1 scene(s) [with extra_splits (239 frames): 1 scene(s)]
Queue 0 Workers 0 Passes 2
Params: --threads=8 --cpu-used=6 --end-usage=q --cq-level=30 --tile-columns=2 --tile-rows=1
⠁ 00:00:00 [---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------]   0%  0/67916 (0 fps, eta 0s)thread '<unnamed>' panicked at 'assertion failed: self.total_chunks != 0', av1an-core/src/broker.rs:116:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Any { .. }', av1an-core/src/settings.rs:1241:21
WARN [av1an_core::ffmpeg] FFmpeg failed to encode audio!
Output {
    status: ExitStatus(
        unix_wait_status(
            256,
        ),
    ),
    stdout: "",
    stderr: "av_interleaved_write_frame(): Input/output error\nError writing trailer of .389f999/audio.mkv: Input/output error\n",
}
Params: "ffmpeg" "-y" "-hide_banner" "-loglevel" "error" "-i" "file.mkv" "-map_metadata" "0" "-vn" "-map" "0" "-map" "-0:a" "-c" "copy" "-map" "0:a" "-c:a" "copy" ".389f999/audio.mkv"
⠁ 00:00:20 [---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------]   0%  0/67916 (0 fps, eta 0s)
```

**Comment by gidoBOSSftw5731** [comment]
That last test was over an NFS share, this is on a local tmpfs (mostly to make testing quicker, also trimmed the subtitle errors as I don't need those right now)

And no, neither lsmash nor ffsm2 are installed, should I attempt to install them? They aren't in the debian repos.

```
$ cat /tmp/foo |grep -v -e 'matroska,webm' -e probesize
INFO [av1an_core::settings] scenecut: found 1 scene(s) [with extra_splits (239 frames): 1 scene(s)]
Queue 0 Workers 0 Passes 2
Params: --threads=8 --cpu-used=6 --end-usage=q --cq-level=30 --tile-columns=2 --tile-rows=1
thread '<unnamed>' panicked at 'assertion failed: self.total_chunks != 0', av1an-core/src/broker.rs:116:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
   2: core::panicking::panic
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:48:5
   3: av1an_core::broker::Broker::encoding_loop
   4: core::ops::function::FnOnce::call_once{{vtable.shim}}
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Any { .. }', av1an-core/src/settings.rs:1241:21
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
   2: core::result::unwrap_failed
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/result.rs:1690:5
   3: av1an_core::settings::EncodeArgs::encode_file::{{closure}}
   4: av1an_core::settings::EncodeArgs::encode_file
   5: av1an_cli::run
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

```

**Comment by redzic** [comment]
I think I know what the problem is. After #598, that changed how frame indexes were handled and thus broke something called extra splits in the case of no scenecuts.

**Comment by gidoBOSSftw5731** [comment]
Is there a solution to this? Should I just pull the commit pre-merge and compile it?

**Comment by redzic** [comment]
@gidoBOSSftw5731  I just made PR #607 that fixes this issue, although I can't really tell if your original error is related since you did not seem to use split method none for that.

**Comment by gidoBOSSftw5731** [comment]
I moved to split-none for later tests as it saves a *lot* of time testing this, and I wanted to try to eliminate variables. I'll pull #607 in a few and see if that changes anything.

**Comment by gidoBOSSftw5731** [comment]
However, on the master branch version, this command did seem to work (I cut it short since I don't want to use AV1, as it's way too intensive for what I do and not yet widely supported)

```
$ RUST_BACKTRACE=1 av1an -i 'file.mkv' -o 'out.mkv' 2>&1 |tee /tmp/foo
[matroska,webm @ 0x560928ac9d00] Could not find codec parameters for stream 5 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928ac9d00] Could not find codec parameters for stream 6 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928ac9d00] Could not find codec parameters for stream 7 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928ac9d00] Could not find codec parameters for stream 8 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928ac9d00] Could not find codec parameters for stream 9 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928add5c0] Could not find codec parameters for stream 5 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928add5c0] Could not find codec parameters for stream 6 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928add5c0] Could not find codec parameters for stream 7 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928add5c0] Could not find codec parameters for stream 8 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928add5c0] Could not find codec parameters for stream 9 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928b5e440] Could not find codec parameters for stream 5 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928b5e440] Could not find codec parameters for stream 6 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928b5e440] Could not find codec parameters for stream 7 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928b5e440] Could not find codec parameters for stream 8 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x560928b5e440] Could not find codec parameters for stream 9 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
Scene detection

INFO [av1an_core::settings] scenecut: found 360 scene(s) [with extra_splits (239 frames): 478 scene(s)]Queue 477 Workers 4 Passes 2
Params: --threads=8 --cpu-used=6 --end-usage=q --cq-level=30 --tile-columns=2 --tile-rows=1
```

**Comment by redzic** [comment]
Reopening as the PR linked only partially fixed this issue to my knowledge.

**Comment by redzic** [comment]
Could you try the original command again on the newest master and post the entire log file just to make sure?

**Comment by gidoBOSSftw5731** [comment]
Doing so now, will report back probably within 30 mins

**Comment by redzic** [comment]
> (scene detection only goes at about 1x speed sadly, maybe I should consider your lsmash or whatever)

lsmash will help speed up piping to the encoder during actual encoding, but it won't speed up scene detection sadly. However there are options related to scene detection speed such as `--sc-downscale-height`, `--sc-pix-format`, and `--sc-method`. So you could try something like `--sc-method=fast --sc-downscale-height=720` and that should be much faster compared to the default options (which are the slowest but also the most accurate).

**Comment by gidoBOSSftw5731** [comment]
> Could you try the original command again on the newest master and post the entire log file just to make sure?

Not fixed, sadly, still the YUV2MPEG2 error. Error also does occur with split-method none, so that helps with testing. That being said, your options do about halve the time taken, which is helpful. 

Command run:
```
$ RUST_BACKTRACE=1 av1an -i 'file.mkv' -e vpx -w 12 -a " -c:a copy -map 0" -v " --cq-level=30 --end-usage=cq --threads=1 "C -o 'out.mkv' --split-method=none
```

**Comment by gidoBOSSftw5731** [comment]
This command does work, though I need to adjust the settings so each worker only uses one thread:
```
RUST_BACKTRACE=1 av1an -i 'file.mkv' -e x265 -o 'out.mkv' -w 12 -a " -c:a copy -map 0 " --sc-method=fast --sc-downscale-height=720 --concat mkvmerge
```

And I still really would prefer vpx

**Comment by redzic** [comment]
Can you try both of these commands and see if either of them work?

```
av1an -i 'file.mkv' -e vpx -w 12 -v "--cq-level=30 --end-usage=cq --threads=1 --bit-depth=10 --profile=2" -o 'out.mkv' --split-method=none --pix-format yuv420p10le
```

```
av1an -i 'file.mkv' -e vpx -w 12 -v "--cq-level=30 --end-usage=cq --threads=1" -o 'out.mkv' --split-method=none --pix-format yuv420p
```

**Comment by gidoBOSSftw5731** [comment]
Command 1 fail:
```
$ av1an -i 'file.mkv' -e vpx -w 12 -v "--cq-level=30 --end-usage=cq --threads=1 --bit-depth=10 --profile=2" -o 'out.mkv' --split-method=none --pix-format yuv420p10le --sc-method=fast --sc-downscale-height=720
[matroska,webm @ 0x55b39cf35f80] Could not find codec parameters for stream 5 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55b39cf35f80] Could not find codec parameters for stream 6 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55b39cf35f80] Could not find codec parameters for stream 7 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55b39cf35f80] Could not find codec parameters for stream 8 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55b39cf35f80] Could not find codec parameters for stream 9 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55b39cf35f80] Could not find codec parameters for stream 5 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55b39cf35f80] Could not find codec parameters for stream 6 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55b39cf35f80] Could not find codec parameters for stream 7 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55b39cf35f80] Could not find codec parameters for stream 8 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[matroska,webm @ 0x55b39cf35f80] Could not find codec parameters for stream 9 (Subtitle: hdmv_pgs_subtitle (pgssub)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
'--bit-depth' isn't a valid parameter for vpx
        Did you mean '--bias-pct'?

To continue anyway, run av1an with '--force'
```

Command 2 is running, a lot of vpxenc threads using a lot of cpu, but nothing yet in the progress bar

**Comment by redzic** [comment]
Does your build of vpxenc somehow not have `--bit-depth` as an option in the `--help`? You could try `--force` as was suggested if you are sure that it does have it but is not documented in the `--help` output.

The output of your other command seems to be failing for strange reasons, specifically the ffmpeg C api is returning some errors on your input file for things like getting the frame rate and checking whether the input video has an audio track. Is there something wrong with your filesystem? You mentioned in an earlier comment that you were running Av1an on an NFS share, could that possibly be somehow causing problems? If an NFS share causes some kind of significant delay between creating files and when you can read them then I could see how this situation could happen, although that doesn't really explain why it's failing to read your input file again unless that's also being ultimately caused by the NFS share.

**Comment by gidoBOSSftw5731** [comment]
Here's my vpxenc help menu, if that helps:
```
$ vpxenc --help 
Usage: vpxenc <options> -o dst_filename src_filename 

Options:
            --help                     	Show usage options and exit
  -D,       --debug                    	Debug mode (makes output deterministic)
  -o <arg>, --output=<arg>             	Output filename
            --codec=<arg>              	Codec to use
  -p <arg>, --passes=<arg>             	Number of passes (1/2)
            --pass=<arg>               	Pass to execute (1/2)
            --fpf=<arg>                	First pass statistics file name
            --limit=<arg>              	Stop encoding after n input frames
            --skip=<arg>               	Skip the first n input frames
  -d <arg>, --deadline=<arg>           	Deadline per frame (usec)
            --best                     	Use Best Quality Deadline
            --good                     	Use Good Quality Deadline
            --rt                       	Use Realtime Quality Deadline
  -q,       --quiet                    	Do not print encode progress
  -v,       --verbose                  	Show encoder parameters
            --psnr                     	Show PSNR in status line
            --webm                     	Output WebM (default when WebM IO is enabled)
            --ivf                      	Output IVF
  -P,       --output-partitions        	Makes encoder output partitions. Requires IVF output!
            --q-hist=<arg>             	Show quantizer histogram (n-buckets)
            --rate-hist=<arg>          	Show rate histogram (n-buckets)
            --disable-warnings         	Disable warnings about potentially incorrect encode settings.
  -y,       --disable-warning-prompt   	Display warnings, but do not prompt user to continue.
            --test-decode=<arg>        	Test encode/decode mismatch
                                       	  off, fatal, warn

Encoder Global Options:
            --yv12                     	Input file is YV12 
            --i420                     	Input file is I420 (default)
            --i422                     	Input file is I422
            --i444                     	Input file is I444
            --i440                     	Input file is I440
  -u <arg>, --usage=<arg>              	Usage profile number to use
  -t <arg>, --threads=<arg>            	Max number of threads to use
            --profile=<arg>            	Bitstream profile number to use
  -w <arg>, --width=<arg>              	Frame width
  -h <arg>, --height=<arg>             	Frame height
            --stereo-mode=<arg>        	Stereo 3D video format
                                       	  mono, left-right, bottom-top, top-bottom, right-left
            --timebase=<arg>           	Output timestamp precision (fractional seconds)
            --fps=<arg>                	Stream frame rate (rate/scale)
            --error-resilient=<arg>    	Enable error resiliency features
            --lag-in-frames=<arg>      	Max number of frames to lag

Rate Control Options:
            --drop-frame=<arg>         	Temporal resampling threshold (buf %)
            --resize-allowed=<arg>     	Spatial resampling enabled (bool)
            --resize-width=<arg>       	Width of encoded frame
            --resize-height=<arg>      	Height of encoded frame
            --resize-up=<arg>          	Upscale threshold (buf %)
            --resize-down=<arg>        	Downscale threshold (buf %)
            --end-usage=<arg>          	Rate control mode
                                       	  vbr, cbr, cq, q
            --target-bitrate=<arg>     	Bitrate (kbps)
            --min-q=<arg>              	Minimum (best) quantizer
            --max-q=<arg>              	Maximum (worst) quantizer
            --undershoot-pct=<arg>     	Datarate undershoot (min) target (%)
            --overshoot-pct=<arg>      	Datarate overshoot (max) target (%)
            --buf-sz=<arg>             	Client buffer size (ms)
            --buf-initial-sz=<arg>     	Client initial buffer size (ms)
            --buf-optimal-sz=<arg>     	Client optimal buffer size (ms)

Twopass Rate Control Options:
            --bias-pct=<arg>           	CBR/VBR bias (0=CBR, 100=VBR)
            --minsection-pct=<arg>     	GOP min bitrate (% of target)
            --maxsection-pct=<arg>     	GOP max bitrate (% of target)
            --corpus-complexity=<arg>  	corpus vbr complexity midpoint

Keyframe Placement Options:
            --kf-min-dist=<arg>        	Minimum keyframe interval (frames)
            --kf-max-dist=<arg>        	Maximum keyframe interval (frames)
            --disable-kf               	Disable keyframe placement

VP8 Specific Options:
            --cpu-used=<arg>           	CPU Used (-16..16)
            --auto-alt-ref=<arg>       	Enable automatic alt reference frames. (0..1)
            --noise-sensitivity=<arg>  	Noise sensitivity (frames to blur)
            --sharpness=<arg>          	Increase sharpness at the expense of lower PSNR. (0..7)
            --static-thresh=<arg>      	Motion detection threshold
            --token-parts=<arg>        	Number of token partitions to use, log2
            --arnr-maxframes=<arg>     	AltRef max frames (0..15)
            --arnr-strength=<arg>      	AltRef filter strength (0..6)
            --arnr-type=<arg>          	AltRef filter type (1..3)
            --tune=<arg>               	Material to favor
                                       	  psnr, ssim
            --cq-level=<arg>           	Constant/Constrained Quality level
            --max-intra-rate=<arg>     	Max I-frame bitrate (pct)
            --gf-cbr-boost=<arg>       	Boost for Golden Frame in CBR mode (pct)
            --screen-content-mode=<arg 	Screen content mode

VP9 Specific Options:
            --cpu-used=<arg>           	CPU Used (-9..9)
            --auto-alt-ref=<arg>       	Enable automatic alt reference frames, 2+ enables multi-layer. (0..6)
            --sharpness=<arg>          	Increase sharpness at the expense of lower PSNR. (0..7)
            --static-thresh=<arg>      	Motion detection threshold
            --tile-columns=<arg>       	Number of tile columns to use, log2
            --tile-rows=<arg>          	Number of tile rows to use, log2 (set to 0 while threads > 1)
            --enable-tpl=<arg>         	Enable temporal dependency model
            --arnr-maxframes=<arg>     	AltRef max frames (0..15)
            --arnr-strength=<arg>      	AltRef filter strength (0..6)
            --arnr-type=<arg>          	AltRef filter type (1..3)
            --tune=<arg>               	Material to favor
                                       	  psnr, ssim
            --cq-level=<arg>           	Constant/Constrained Quality level
            --max-intra-rate=<arg>     	Max I-frame bitrate (pct)
            --max-inter-rate=<arg>     	Max P-frame bitrate (pct)
            --gf-cbr-boost=<arg>       	Boost for Golden Frame in CBR mode (pct)
            --lossless=<arg>           	Lossless mode (0: false (default), 1: true)
            --frame-parallel=<arg>     	Enable frame parallel decodability features
            --aq-mode=<arg>            	Adaptive quantization mode (0: off (default), 1: variance 2: complexity, 3: cyclic refresh, 4: equator360)
            --alt-ref-aq=<arg>         	Special adaptive quantization for the alternate reference frames.
            --frame-boost=<arg>        	Enable frame periodic boost (0: off (default), 1: on)
            --noise-sensitivity=<arg>  	Noise sensitivity (frames to blur)
            --tune-content=<arg>       	Tune content type
                                       	  default, screen, film
            --color-space=<arg>        	The color space of input content:
                                       	  unknown, bt601, bt709, smpte170, smpte240, bt2020, reserved, sRGB
            --min-gf-interval=<arg>    	min gf/arf frame interval (default 0, indicating in-built behavior)
            --max-gf-interval=<arg>    	max gf/arf frame interval (default 0, indicating in-built behavior)
            --target-level=<arg>       	Target level
                                        255: off (default)
                                          0: only keep level stats
                                          1: adaptively set alt-ref distance and column tile limit based on picture size, and keep level stats
                                         10: level 1.0  11: level 1.1  ...  62: level 6.2
            --row-mt=<arg>             	Enable row based non-deterministic multi-threading in VP9

Stream timebase (--timebase):
  The desired precision of timestamps in the output, expressed
  in fractional seconds. Default is 1/1000.

Included encoders:

    vp8    - WebM Project VP8 Encoder v1.8.1 
    vp9    - WebM Project VP9 Encoder v1.8.1 (default)

        Use --codec to switch to a non-default encoder.

```


And all tests after the first one are NOT on an NFS share. They are all now on a tmpfs share, just to increase speed and minimize bottlenecks.

The second command is running and after processing ~1200 frames it's now at the whopping speed of 1.18fps! At this rate, that's not significantly faster than FFMPEG, so the speed itself isn't necessarily a problem, but ffmpeg has the advantage of being more versatile and having more tools for automating (as I want this to do a pass over a large amount of my library at some point.)

**Comment by gidoBOSSftw5731** [comment]
Ok, update on the second command, I just restarted it (had to reboot the box then was out for the night) I installed ffsm2 and it's now at 1200frames at 1.4fps, so there's that. I also moved back to the hard drives but not over NFS, just because I couldn't be arsed to copy it back over and wait another 10 minutes.

It seems to be running smoothly for now, I'll keep you updated on how it ends up in 5 or so hours. It starts really slowly but vspipe seems to be working hard at that point shoveling data somewhere, because after all the VSPipe threads settle down it moves quicker. I also ended a compute task on 6 of my 24 threads, so that could be helping to speed it up.

**Comment by gidoBOSSftw5731** [comment]
Ok, to confirm, this specific command finished correctly, and it took 4h40m to complete at an average of 1.59fps. `RUST_BACKTRACE=1 av1an -i 'file.mkv' -e vpx -w 12 -v "--cq-level=30 --end-usage=cq --threads=1" -o 'out.mkv' --split-method=none --pix-format yuv420p --sc-method=fast --sc-downscale-height=720 -m ffms2`

Will try again with full scene checking. I'm more than happy to take feedback on how to make it faster, but this is working and that's valuable. I'm also glad it kept the subtitles and all audio tracks.

Closing this unless I find another issue. Thanks for the help
